packages/node_modules/@webex/plugin-meetings/test/unit/spec/locus-info/index.js

```diff
@@ -828,6 +828,52 @@ describe('plugin-meetings', () => {
       });
     });
 
+    describe('#updateEmbeddedApps()', () => {
+      const newEmbeddedApps = [{
+        url: 'https://hecate-b.wbx2.com/apps/api/v1/locus/7a4994a7',
+        sequence: 138849877016800000,
+        appId: 'Y2lzY29zcGFyazovL3VzL0FQUExJQ0FUSU9OLzQxODc1MGQ0LTM3ZDctNGY2MC1hOWE3LWEwZTE1NDFhNjRkNg',
+        instanceInfo: {
+          appInstanceUrl: 'https://webex.sli.do/participant/event/mFKKjcYxzx9h31eyWgngFS?clusterId=eu1',
+          externalAppInstanceUrl: '',
+          title: 'Active session'
+        },
+        state: 'STARTED',
+        lastModified: '2022-10-13T21:01:41.680Z'
+      }];
+
+      it('updates the embeddedApps object', () => {
+        const prev = locusInfo.embeddedApps;
+
+        locusInfo.updateEmbeddedApps(newEmbeddedApps);
+
+        assert.notEqual(locusInfo.embeddedApps, prev);
+      });
+
+      it('emits EMBEDDED_APPS_UPDATED when apps changed', () => {
+        locusInfo.emitScoped = sinon.stub();
+
+        locusInfo.updateEmbeddedApps(newEmbeddedApps);
+
+        assert.calledWith(locusInfo.emitScoped, {
+          file: 'locus-info',
+          function: 'updateEmbeddedApps'
+        },
+        LOCUSINFO.EVENTS.EMBEDDED_APPS_UPDATED,
+        newEmbeddedApps);
+      });
+
+      it('does not emit EMBEDDED_APPS_UPDATED when apps didn\'t change', () => {
+        locusInfo.updateEmbeddedApps(newEmbeddedApps);
+
+        locusInfo.emitScoped = sinon.stub();
+
+        locusInfo.updateEmbeddedApps(newEmbeddedApps);
```

Comment:
i'm not sure this test does exactly what i think it should. in this test we're comparing the same reference to `newEmbeddedApps` so it doesn't fire a new event. would it be a more robust test to have a line between 867 and 871 that is `const clonedApps = deepClone(newEmbeddedApps)` and then pass that to `updateEmbeddedApps`?

Reply:
`updateEmbeddedApps` already makes a clone of the embeddedApps object, so passing in the same `newEmbeddedApps` should test it just fine.  But if someone changes the implementation in the future it might not test it properly, so I think your suggestion is better.